services/keycloak/start.sh

```diff
@@ -1293,6 +1293,31 @@ function add_group_viewall {
 EOF
 }
 
+function configure_task_cron {
+
+  CLIENT_ID=$(/opt/jboss/keycloak/bin/kcadm.sh get -r lagoon clients?clientId=api --config $CONFIG_PATH | python -c 'import sys, json; print json.load(sys.stdin)[0]["id"]')
+  cron_task_groups=$(/opt/jboss/keycloak/bin/kcadm.sh get -r lagoon clients/$CLIENT_ID/authz/resource-server/permission?name=Run+Drush+cron --config $CONFIG_PATH)
+
+  if [ "$cron_task_groups" != "[ ]" ]; then
+    echo "group:drushCron already configured"
+    return 0
+  fi
+
+  echo Configuring group:drushCron
+
+  /opt/jboss/keycloak/bin/kcadm.sh create clients/$CLIENT_ID/authz/resource-server/permission/scope --config $CONFIG_PATH -r lagoon -f - <<EOF
+{
+  "name": "Run Drush cron",
+  "type": "scope",
+  "logic": "POSITIVE",
+  "decisionStrategy": "UNANIMOUS",
+  "resources": ["task"],
+  "scopes": ["drushCron:production","drushCron:development"],
```

Comment:
Thanks for moving this. Unfortunately, this will still fail in an upgrade path since the two new scopes will only be added during a fresh install. And by "fail" I mean the install script will complete w/o errors, but evaluating these permissions will always error (and return permission denied).

In this function you'll need to add the two new scopes (can copy the `# Resource Scopes` section line ~159), and then add those scopes to the `task` resource. For an example of that you can look at my PR for UI improvements. Unfortunately you have to send a PATCH request will all scopes, not just the ones that need adding: https://github.com/amazeeio/lagoon/pull/1165/files#diff-8c9af22e910ccf7cd25a29a92c5faadaR1307-R1308

When those are moved, it'd be best to remove them from `configure_api_client` also that way there won't be any errors on a fresh install.

Reply:
@rocketeerbkw,

I believe I've tended to what you've described.

Thanks for your patience.

Reply:
Sorry this update mechanism isn't very intuitive. I've fix it in https://github.com/amazeeio/lagoon/commit/a39d6d2c36278da1a7457498b2865d39bbd756d1, can you merge that in? Should be good to go after that

Reply:
I have cherry-picked a39d6d2c36278da1a7457498b2865d39bbd756d1